### PR TITLE
Removed the any from the scales

### PR DESCRIPTION
**Pull request by aicioara**
Also removed the coercer and no longer accepting coercing (for example passing numbers formatted as strings when numbers are required)

### CODE REVIEW

**Comment by aicioara** [comment]
Self reviewing for a second

File: test/scales/timeScaleTests.ts [hunk 1]
```diff
@@ -15,24 +15,14 @@ describe("TimeScale tests", () => {
       var time2 = scale.domain()[1].valueOf();
       assert.strictEqual(time2, secondDate, "first value of domain set correctly");
     }
-    checkDomain(["10/1/2014", "11/1/2014"]);
-    checkDomain(["October 1, 2014", "November 1, 2014"]);
-    checkDomain(["Oct 1, 2014", "Nov 1, 2014"]);
+    checkDomain([new Date("10/1/2014"), new Date("11/1/2014")]);
+    checkDomain([new Date("October 1, 2014"), new Date("November 1, 2014")]);
+    checkDomain([new Date("Oct 1, 2014"), new Date("Nov 1, 2014")]);
```
Comment (aicioara) [inline test/scales/timeScaleTests.ts]: Unsure if we still want this test. It is actually checking JS coercing behavior (which we should assume correct)

Comment (jtlan) [inline test/scales/timeScaleTests.ts]: Yeah, if we don't have the coercer, there's no point in testing this.

**Comment by jtlan** [comment]
I like it! I think we should also make sure that users downstream aren't going to be strongly affected by the removal of this functionality. @crmorford, @istarnx, any thoughts?

**Comment by jtlan** [comment]
@aicioara Please fix any Overlaying Tests this change may have broken. A number of them seem to depend on the `string` coercer logic on `Scales.Category()`.

**Comment by istarnx** [comment]
@jtlan I scanned our code and don't think this'll be painful for us. Is there anything I'm missing? What's the major user impact here?

**Comment by aicioara** [comment]
Fixing overlaying tests

File: src/scales/quantitativeScale.ts [hunk 2]
```diff
@@ -115,8 +114,8 @@ module Plottable {
       }
     }
 
-    public _defaultExtent(): any[] {
-      return [0, 1];
+    public _defaultExtent(): D[] {
+      throw Error("The quantitative scale itself does not have a default extent");
```
Comment (bluong) [inline src/scales/quantitativeScale.ts]: This sorta basically disallows QuantitativeScales to be instantiated and used on its own.  Is this ok?

Comment (aicioara) [inline src/scales/quantitativeScale.ts]: This is a good point. Now the problem is: 
1. We want to remove `any` everywhere it is possible. 
2. There is no default extent when there is no concrete type.

Also, a sign that there is a problem is the underscore in the name. I suggest we remove this method altogether. It is only used in the `domainer.ts`

Comment (aicioara) [inline src/scales/quantitativeScale.ts]: opened https://github.com/palantir/plottable/issues/2015 for this issue

**Pull request merged**